FlowCrypt/Controllers/SideMenu/Main/SideMenuNavigationController.swift

```diff
@@ -114,38 +114,39 @@ extension SideMenuNavigationController {
                 }
                 let emailKeyManagerApi = EmailKeyManagerApi(clientConfiguration: configuration)
                 let idToken = try await IdTokenUtils.getIdToken(userEmail: context.user.email)
-                let keydDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
+                let keyDetails = try await emailKeyManagerApi.getPrivateKeys(idToken: idToken)
                 let localKeys = try context.encryptedStorage.getKeypairs(by: context.user.email)
-                var savedPassPhrase = localKeys.first(where: { $0.passphrase != nil })?.passphrase
-                var isKeyUpdated = false
-                for keyDetail in keydDetails {
-                    guard let savedLocalKey = localKeys.first(where: { $0.primaryFingerprint == keyDetail.primaryFingerprint }) else {
-                        // No keys found in local. Add it
-                        savedPassPhrase = try await saveKeyToLocal(
-                            context: context,
-                            keyDetail: keyDetail,
-                            passPhrase: savedPassPhrase,
-                            isNewKey: true
-                        )
-                        isKeyUpdated = true
-                        continue
-                    }
-                    guard let lastModified = keyDetail.lastModified else {
-                        continue
-                    }
-                    // Key exists in local. Check if saved key is outdated by checking lastModified and update if needed
-                    if savedLocalKey.lastModified ?? 0 < lastModified {
-                        savedPassPhrase = try await saveKeyToLocal(
-                            context: context,
-                            keyDetail: keyDetail,
-                            passPhrase: savedPassPhrase
-                        )
-                        isKeyUpdated = true
+
+                var passPhrase = localKeys.first(where: { $0.passphrase != nil })?.passphrase
+                // If this is called when starting the app, then it doesn't make much difference
+                // but conceptually it would be better to look pass phrase both in memory and storage
+                if passPhrase == nil {
+                    passPhrase = try context.passPhraseService.getPassPhrases().first(where: { $0.value.isNotEmpty })?.value
```

Comment:
My understanding was that `context.passPhraseService.getPassPhrases()` already looked in both storage and memory. Could you check? Then you'd only need that one.

Reply:
Yeah, indeed it looks in both storage.
But  `encryptedStorage.getPassPhrases()` function doesn't search pass phrases by user email.
What I worry is it might return incorrect pass phrase (when user have 2 acccounts (`A`, `B`), it might return `A` private key for `B` account)

What do you think?
```
func getPassPhrases() throws -> [PassPhrase] {
    try encryptedStorage.getPassPhrases() + inMemoryStorage.getPassPhrases()
}
```

Reply:
Wow, seriously? There is no "current account" in the init or somewhere either? Or init of encrypted storage or inMemoryStorage?

Reply:
This is `encryptedStorage.getPassPhrases` function and I can't find user email check.

```
func getPassPhrases() throws -> [PassPhrase] {
    return try storage.objects(KeypairRealmObject.self)
        .compactMap(PassPhrase.init)
}
```

Reply:
Ok. That needs to be fixed for sure. Well noticed.

Reply:
then how about we fix in this PR?

Reply:
as i have to use getPassPhrases function in this PR as well?
Or do you wanna to address in new issue?

Reply:
@tomholub

Reply:
sure, let's fix it here